### PR TITLE
[Authorization] Fix for issue #40050

### PR DESCRIPTION
**Pull request by melina5656**
Fixed error in GetAllAsync().

### CODE REVIEW

**Comment by HarveyLink** [comment]
This is a fix PR for issue: #40050

File: sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        [Test]
+        public async Task Get()
+        {
+            var randomGuid = Recording.Random.NewGuid().ToString();
+            var scope = DefaultSubscription.Id;
+            var denyAssignmentCollection = Client.GetDenyAssignments(scope);
+            try
+            {
+                var denyAssignment = (await denyAssignmentCollection.GetAsync(randomGuid)).Value;
+                Assert.AreEqual(1, 0);
```
Comment (live1206) [inline sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs:31]: what is this always fail assertion for?

File: sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            }
+            catch (RequestFailedException ex) when (ex.Status == 404)
+            {
+                Assert.AreEqual(1, 1);
```
Comment (live1206) [inline sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs:35]: what is this always true assertion for?

File: sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            var randomGuid = Recording.Random.NewGuid().ToString();
+            var scope = DefaultSubscription.Id;
+            var denyAssignmentCollection = Client.GetDenyAssignments(scope);
+            try
+            {
+                var denyAssignment = (await denyAssignmentCollection.GetAsync(randomGuid)).Value;
+                Assert.AreEqual(1, 0);
+            }
+            catch (RequestFailedException ex) when (ex.Status == 404)
```
Comment (live1206) [inline sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs:33]: use `Assert.Throws<RequestFailedException>` to verify the expecting exception

File: sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            }
+        }
+
+        [Test]
+        public async Task GetAllBySubscription()
+        {
+            var scope = DefaultSubscription.Id;
+            var denyAssignmentCollection = Client.GetDenyAssignments(scope);
+            var denyAssignments = await denyAssignmentCollection.GetAllAsync().ToEnumerableAsync();
+            Assert.AreEqual(0, denyAssignments.Count);
```
Comment (live1206) [inline sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs:45]: use `Assert.IsEmpty(denyAssignments)` instead, same for the other tests

Comment (melina5656) [reply sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs]: Resolved

File: sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs (hunk 3 above)
Comment (melina5656) [reply sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs]: Resolved

**Review by live1206 — APPROVED** [review]

File: sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs (hunk 1 above)
Comment (melina5656) [reply sdk/authorization/Azure.ResourceManager.Authorization/tests/Scenario/DenyAssignmentCollectionTests.cs]: If I want to test the get method, there must be an existing `denyAssignmentId`, but the list of DenyAssignments in the Azure portal is `empty`, so I cannot obtain the `denyAssignmentId`. Therefore, I can only test whether the get method will pass a non-existent Guid.

`Assert AreEqual (1, 0); `The function of is to terminate the program.
`Assert AreEqual (1, 1); `It is used to ensure that the program continues to run after obtaining the 404 error message.

Now I have deleted these two pieces of code and changed to a new format.

**Pull request merged**